dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -1864,8 +2149,8 @@ private static void update(Map<Event, List<Event>> map, Collection<Tuple> set) {
         }
     }
 
-    private boolean implies(Event y, Event z) {
-        return exec.isImplied(z, y);
+    private boolean implies(Event premised, Event concluded) {
+        return exec.isImplied(concluded, premised);
```

Comment:
I think the order is wrong: `exec.isImplied(x, y)` already means `exec(x) => exec(y)`.